secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v4_0_0/domesticscheduledpayments/DomesticScheduledPaymentsApiController.java

```diff
@@ -0,0 +1,280 @@
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.ConsumePaymentConsentRequest;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.domesticscheduled.v3_1_10.DomesticScheduledPaymentConsent;
+import com.forgerock.sapi.gateway.rs.resource.store.repo.entity.payment.FRDomesticScheduledPaymentSubmission;
+import com.forgerock.sapi.gateway.rs.resource.store.repo.mongo.payments.DomesticScheduledPaymentSubmissionRepository;
+import com.forgerock.sapi.gateway.uk.common.shared.api.meta.obie.OBVersion;
+
+import jakarta.servlet.http.HttpServletRequest;
+import uk.org.openbanking.datamodel.v4.common.Meta;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteDomesticScheduled2;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteDomesticScheduledResponse5;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteDomesticScheduledResponse5Data;
+import uk.org.openbanking.datamodel.v4.payment.OBWriteDomesticStandingOrderResponse6Data;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetails1;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetails1PaymentStatusStatus;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetails1Status;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetails1StatusDetail;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetails1StatusDetailStatus;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetailsResponse1;
+import uk.org.openbanking.datamodel.v4.payment.OBWritePaymentDetailsResponse1Data;
+
+@Controller("DomesticScheduledPaymentsApiV4.0.0")
+public class DomesticScheduledPaymentsApiController implements DomesticScheduledPaymentsApi {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final DomesticScheduledPaymentSubmissionRepository scheduledPaymentSubmissionRepository;
+    private final PaymentSubmissionValidator paymentSubmissionValidator;
+    private final DomesticScheduledPaymentConsentStoreClient consentStoreClient;
+    private final OBValidationService<OBWriteDomesticScheduled2ValidationContext> paymentValidator;
+    private final RefundAccountService refundAccountService;
+    private final IdempotentPaymentService<FRDomesticScheduledPaymentSubmission, FRWriteDomesticScheduled> idempotentPaymentService;
+
+    public DomesticScheduledPaymentsApiController(
+            DomesticScheduledPaymentSubmissionRepository paymentSubmissionRepository,
+            PaymentSubmissionValidator paymentSubmissionValidator,
+            @Qualifier("v4.0.0RestDomesticScheduledPaymentConsentStoreClient") DomesticScheduledPaymentConsentStoreClient consentStoreClient,
+            RefundAccountService refundAccountService,
+            OBValidationService<OBWriteDomesticScheduled2ValidationContext> paymentValidator
+    ) {
+        this.scheduledPaymentSubmissionRepository = paymentSubmissionRepository;
```

Comment:
null-checks on these fields?

Reply:
Already checked in the service, consentStore, etc, classes in both RS and RCS.